### PR TITLE
gitbutler 0.10.6 (new cask)

### PR DESCRIPTION
**Pull request by SeanSith**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Review by bevanjkay — DISMISSED** [review]

File: Casks/g/gitbutler.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+cask "gitbutler" do
+  arch arm: "aarch64", intel: "x86_64"
+
+  version "0.10.6"
+  sha256 arm:   "9556c2aff50b3f1372828f067019179aa3e9b6ac3a88ee20ec837273b2202fe3",
+         intel: "ac5908a954b384de21484472d7305da7c680776e6beddd84acd59674ad0c28c9"
+
+  url "https://app.gitbutler.com/downloads/release/darwin/#{arch}/dmg"
```
[the remark this answers is not in the document]
Comment (SeanSith) [reply Casks/g/gitbutler.rb:8]: I guess this is where I could use some guidance. I can glean a versioned URL from the vendor, but it's complicated based on this [comment I made in the app's repo](https://github.com/gitbutlerapp/gitbutler/issues/2593#issuecomment-1941848519). What would Homebrew's maintainers prefer in this particular instance?

[the remark this answers is not in the document]
Comment (bevanjkay) [reply Casks/g/gitbutler.rb:8]: There's probably a way to make it work. What is the versioned url you found?

[the remark this answers is not in the document]
Comment (krehel) [reply Casks/g/gitbutler.rb:8]: @SeanSith there's no real issue with that IMHO.  The cost of coding that is one time (which you've already done).  

We would prefer a versioned URL over something not versioned.  That being said, we also want something that is consistent (which you reference in your comment as well, i.e. don't flip between `x86_64` and `x64`).  If the scheme you have in that comment can more or less be surmised to remain that way, it's fine.

Regarding build numbers, we can include those as well as a second part of the version.  But if it's not needed to fetch or identify an artifact, then probably best to not include it.

[the remark this answers is not in the document]
Comment (bevanjkay) [reply Casks/g/gitbutler.rb:8]: Can you out the actual url not the interpolated url?

[the remark this answers is not in the document]
Comment (krehel) [reply Casks/g/gitbutler.rb:8]: I rewrote a good portion of this to get it working, but not final state.  It's a bit ugly, but does seem ok.  Mind if I push @SeanSith?

@bevanjkay would like your eyes on it as well, please.

[the remark this answers is not in the document]
Comment (SeanSith) [reply Casks/g/gitbutler.rb]: @krehel I have a fledgling fix for this, as well (pushed, sorry I didn't see your message), except that I had to disable the livecheck as I didn't know how to solve that issue. You're welcome to push your solution if you're further along than I am!

[the remark this answers is not in the document]
Comment (krehel) [reply Casks/g/gitbutler.rb]: No worries @SeanSith.  I figured you were actively working on it so I didn't want to push changes into your branch without commenting.  I'll push, but it's going to overwrite many things you've done.  We'll eventually arrive at consensus. 😃

[the remark this answers is not in the document]
Comment (SeanSith) [reply Casks/g/gitbutler.rb]: @bevanjkay The URLs are:
```
https://releases.gitbutler.com/releases/release/0.10.6-619/macos/aarch64/GitButler_0.10.6_aarch64.dmg
https://releases.gitbutler.com/releases/release/0.10.6-619/macos/x86_64/GitButler_0.10.6_x64.dmg
```

**Comment by SeanSith** [comment]
Added zap and rebased to one commit.

**Comment by krehel** [comment]
Thanks.  It would be ideal if you could remove the need for `on_arch_conditional` as you mentioned in your previous comments with the naming scheme as it's just an unnecessary complication, but this can suffice.  If that change happens, we'll have to obviously make adjustments.

Overall, think this looks pretty good.  Thank you @SeanSith!

**Review by krehel — APPROVED** [review]

**Review by bevanjkay — APPROVED** [review]

**Review by razvanazamfirei — APPROVED** [review]

**Pull request merged**